executors/rust/1.3/src/relativedatetime_fmt.rs

```diff
@@ -132,9 +132,21 @@ pub fn run_relativedatetimeformat_test(json_obj: &Value) -> Result<Value, String
 
     let locale_json_str: &str = json_obj["locale"].as_str().unwrap();
     let mut locale_str: String = locale_json_str.to_string();
+
     if numbering_system_str.is_some() {
-        locale_str =
-            locale_json_str.to_string() + "-u-nu-" + &numbering_system_str.as_ref().unwrap();
+       let numbering_system = numbering_system_str.as_ref().unwrap();
+       locale_str =
+            locale_json_str.to_string() + "-u-nu-" + numbering_system;
```

Comment:
The proper way to modify a locale is not to do string concatenation b/c what if the locale coming in already has that specified?

We should probably use the `icu::locid` crate for inspecting and setting Unicode [programmatically via the API](https://docs.rs/icu/1.3.0/icu/locid/extensions/unicode/struct.Keywords.html).